modules/REST/V1/Options.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V1;
+
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+use WP_Syntex\Polylang\Options\Options as Options_Registry;
+use WP_Syntex\Polylang\REST\Abstract_Controller;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Options REST controller.
+ *
+ * @since 3.7
+ */
+class Options extends Abstract_Controller {
+	/**
+	 * @var Options_Registry
+	 */
+	private $options;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Options_Registry $options Options registry.
+	 */
+	public function __construct( Options_Registry $options ) {
+		$this->namespace = 'pll/v1';
+		$this->rest_base = 'settings';
+		$this->options   = $options;
+	}
+
+	/**
+	 * Registers the routes for options.
+	 *
+	 * @since 3.7
+	 *
+	 * @return self
+	 */
+	public function register_routes(): self {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				'args'        => array(
+					'context' => $this->get_context_param( array( 'default' => 'edit' ) ),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_items' ),
+					'permission_callback' => array( $this, 'update_items_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
+			)
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves all options.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items( $request ) {
+		return $this->prepare_item_for_response( $this->options->get_all(), $request );
+	}
+
+	/**
+	 * Updates option(s).
+	 * This allows to unpdate one or several options.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function update_items( $request ) {
+		$response = array();
+		$errors   = new WP_Error();
+		$schema   = $this->options->get_schema();
+		$options  = array_intersect_key( $this->options->get_all(), $request->get_params() ); // Remove `context`.
+
+		foreach ( $options as $option_name => $current_value ) {
+			if ( ! empty( $schema['properties'][ $option_name ]['readonly'] ) ) {
```

Comment:
Isn't it checked automatically by WordPress?
I cannot found where WP does such a thing in a concrete controller

Reply:
WP does strictly nothing at this point.
The only moment WP uses it is in `rest_get_endpoint_args_for_schema()`.

Reply:
I mean for WordPress core routes, such as for posts.

Reply:
Ha! Haven't tried. They probably use `rest_get_endpoint_args_for_schema()` I suppose.

Reply:
They simply ignore the fields that are `readonly`.

Reply:
For simplicity, should we do the same here?

Reply:
Done in [cf488c7](https://github.com/polylang/polylang/pull/1569/commits/cf488c7434d3c32899152ad4371b784f49e4d2d4).